src/views/preview/comment/comment.jsx

```diff
@@ -105,27 +112,44 @@ class Comment extends React.Component {
                             href={`/users/${author.username}`}
                         >{author.username}</a>
                         <div className="action-list">
-                            {deletable ? (
-                                <span
-                                    className="comment-delete"
-                                    onClick={this.handleDelete}
-                                >
-                                    <FormattedMessage id="comments.delete" />
-                                </span>
-                            ) : null}
-                            <span
-                                className="comment-report"
-                                onClick={this.handleReport}
-                            >
-                                <FormattedMessage id="comments.report" />
-                            </span>
+                            {visible ? (
+                                <React.Fragment>
+                                    {canDelete ? (
+                                        <span
+                                            className="comment-delete"
+                                            onClick={this.handleDelete}
+                                        >
+                                            <FormattedMessage id="comments.delete" />
+                                        </span>
+                                    ) : null}
+                                    <span
+                                        className="comment-report"
+                                        onClick={this.handleReport}
+                                    >
+                                        <FormattedMessage id="comments.report" />
+                                    </span>
+                                </React.Fragment>
+                            ) : (
+                                <React.Fragment>
+                                    <span className="comment-visibility">
+                                        <FormattedMessage id={`comments.status.${visibility}`} />
+                                    </span>
+                                    {canRestore && (
```

Comment:
Would be good to be consistent in how to conditionally render. I like this way better personally.